src/dialect/mssql/mssql-driver.ts

```diff
@@ -84,7 +87,9 @@ export class MssqlDriver implements Driver {
   }
 
   async releaseConnection(connection: MssqlConnection): Promise<void> {
-    await connection[PRIVATE_RELEASE_METHOD]()
+    if (this.#config.tedious.resetConnectionOnRelease !== false) {
+      await connection[PRIVATE_RELEASE_METHOD]()
+    }
```

Comment:
this leaks "lower-level" detail.
let's do it @ `connection[PRIVATE_RELEASE_METHOD]` instead.

Reply:
We would need to make `connection[PRIVATE_RELEASE_METHOD]()` aware of the config option in order to do that. It could be passed as a function argument `connection[PRIVATE_RELEASE_METHOD](this.#config.tedious.resetConnectionOnRelease)`, or it could be passed to the `MssqlConnection` constructor.

Any opinion?

Reply:
Let's pass it to the constructor.